src/main/java/com/swp/user/domain/RelationshipId.java

```diff
@@ -0,0 +1,7 @@
+package com.swp.user.domain;
+
+import java.io.Serializable;
+
+public class RelationshipId implements Serializable {
+    private Long fromUser,toUser;
+}
```

Comment:
이 클래스가 이제 Primary Key를 나타내는 클래스가 되는데, JPA는 primary key를 `equals()`로 비교해서 [이 코드](https://github.com/sang-w0o/wantChu-Server/blob/master/src/main/java/com/wantchu/domain/favorite/CompositeFavoriteKeys.java) 처럼 `equals()` 메소드 오버라이딩 해줘야 해요!
```suggestion
@AllArgsConstructor
@NoArgsConstructor
public class RelationshipId implements Serializable {
    private Long fromUser;
    private Long toUser;
}
```

Reply:
코드가 안 보여요

Reply:
Relationship은 두 키를 모두 사용해 구분하는 복합 키라서 그에 맞게 equals()를 오버라이딩 해줘야겠네요.

[여기](https://docs.jboss.org/hibernate/stable/core.old/reference/en/html/persistent-classes-equalshashcode.html)를 참조해보세요!

```java
@AllArgsConstructor
@NoArgsConstructor
public class RelationshipId implements Serializable {
    private Long fromUser;
    private Long toUser;
    
    @Override
    public boolean equals(Object obj) {
        if(obj == null)
            return false;
        // Todo: this.fromUser, this.toUser <->  ((RelationshipId)obj).fromUser,((RelationshipId)obj).toUser를  모두 비교하는 메소드 작성
        ...
    }
}
```